crates/assembly/src/assembler.rs

```diff
@@ -866,8 +849,9 @@ impl Assembler {
                     self.linker.register_procedure_root(procedure_gid, proc_mast_root)?;
                     mast_forest_builder.insert_procedure(procedure_gid, procedure)?;
                 },
-                // Type and constant exports are irrelevant at this stage
-                Export::Type(_) | Export::Constant(_) => continue,
+                SymbolItem::Compiled(_) | SymbolItem::Constant(_) | SymbolItem::Type(_) => {
+                    unreachable!()
```

Comment:
Same comment as above: could we add a brief explanation why `unreachable!` is OK here?